docs/shared/asm-ingress-gateway/gateway.yaml

```diff
@@ -0,0 +1,47 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: asm-ingressgateway
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: asm-ingressgateway
+spec:
+  type: LoadBalancer
+  selector:
+    asm: ingressgateway
+  ports:
+  - port: 80
+    name: http
+    targetPort: 8080
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: asm-ingressgateway
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      asm: ingressgateway
+  template:
+    metadata:
+      annotations:
+        # This is required to tell Anthos Service Mesh to inject the gateway with the
+        # required configuration.
+        inject.istio.io/templates: gateway
+      labels:
+        asm: ingressgateway
+    spec:
+      containers:
+      - name: istio-proxy
+        image: auto # The image will automatically update each time the pod starts.
+        resources:
+          limits:
+            cpu: 2000m
+            memory: 1024Mi
+          requests:
+            cpu: 100m
+            memory: 128Mi
+      serviceAccountName: asm-ingressgateway
```

Comment:
**Thought:**
The file name "gateway.yaml" made me (at first) think that this file contains a Gateway object.
Should the file name be more specific (e.g., `gateway-deployment.yaml`)?

I was also thinking about splitting but that goes against [this best practice from the official K8s docs](https://kubernetes.io/docs/concepts/configuration/overview/#:~:text=Group%20related%20objects%20into%20a%20single%20file%20whenever%20it%20makes%20sense.%20One%20file%20is%20often%20easier%20to%20manage%20than%20several.%20See%20the%20guestbook%2Dall%2Din%2Done.yaml%20file%20as%20an%20example%20of%20this%20syntax.).
> Group related objects into a single file whenever it makes sense. One file is often easier to manage than several.

Reply:
true, i'll rename